src/ZoneServer/Skills/Handlers/Highlander/CrossCut.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Melia.Shared.Data.Database;
+using Melia.Shared.L10N;
+using Melia.Shared.Game.Const;
+using Melia.Shared.World;
+using Melia.Zone.Network;
+using Melia.Zone.Skills.Combat;
+using Melia.Zone.Skills.Handlers.Base;
+using Melia.Zone.Skills.SplashAreas;
+using Melia.Zone.World.Actors;
+using Melia.Zone.World.Actors.Characters.Components;
+using Melia.Zone.World.Actors.CombatEntities.Components;
+using static Melia.Zone.Skills.SkillUseFunctions;
+
+namespace Melia.Zone.Skills.Handlers.Highlander
+{
+	/// <summary>
+	/// Handler for the Highlander skill Crosscut.
+	/// </summary>
+	[SkillHandler(SkillId.Highlander_CrossCut)]
+	public class CrossCut : IGroundSkillHandler
+	{
+		/// <summary>
+		/// Handles skill, damaging targets.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="originPos"></param>
+		/// <param name="farPos"></param>
+		public void Handle(Skill skill, ICombatEntity caster, Position originPos, Position farPos, ICombatEntity target)
+		{
+			if (!caster.TrySpendSp(skill))
+			{
+				caster.ServerMessage(Localization.Get("Not enough SP."));
+				return;
+			}
+
+			skill.IncreaseOverheat();
+			caster.TurnTowards(farPos);
+			caster.SetAttackState(true);
+
+			var splashParam = skill.GetSplashParameters(caster, originPos, farPos, length: 45, width: 30, angle: 0);
+			var splashArea = skill.GetSplashArea(SplashType.Square, splashParam);
+
+			Send.ZC_SKILL_READY(caster, skill, originPos, farPos);
+			Send.ZC_SKILL_MELEE_GROUND(caster, skill, farPos, null);
+
+			this.Attack(skill, caster, splashArea);
+		}
+
+		/// <summary>
+		/// Executes the actual attack after a delay.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="splashArea"></param>
+		private async void Attack(Skill skill, ICombatEntity caster, ISplashArea splashArea)
+		{
+			var hitDelay = TimeSpan.FromMilliseconds(30);
+			var damageDelay1 = TimeSpan.FromMilliseconds(50);
+			var damageDelay2 = TimeSpan.FromMilliseconds(80);
+			var delayBetweenHits = TimeSpan.FromMilliseconds(330);
+			var skillHitDelay = TimeSpan.Zero;
+
+			var pc = skill.Owner;
```

Comment:
Is there a specific reason why you're using the skill owner instead of the caster here?

Reply:
Haha, it's actually because you wrote that code snippet for me and I just pasted it verbatim.
https://discord.com/channels/273278107773960192/1128408008469266503/1185313478928506930

You're right though, it should be caster.  I tested it with caster and it still works fine.